subprojects/core/src/main/java/org/gradle/execution/plan/DefaultExecutionPlan.java

```diff
@@ -365,6 +365,7 @@ public NodeInVisitingSegment apply(Node node) {
             }
         }
         executionQueue.clear();
+        dependencyResolver.clear();
```

Comment:
What amount of savings are we talking about here?

Reply:
250MB on a 1000 module Android build. It's the biggest individual object in their heap.

Reply:
Is the heap dump available somewhere? I'm just wondering if there's a way to reduce 250MB in question.

Of course, `.clear()` is good, however, it would be nice as well to reduce the initial overhead.

Reply:
The dump is not public unfortunately. I didn't see much room for improvement in that area, the biggest problem was that it was kept around during execution, when it was no longer needed.